### DISCUSSION TITLE
How do I redeploy?

### DISCUSSION THREAD

**Discussion by lete114** [deta/cloud-docs#504] [Help] [answered]
For example, if I deploy an application that may have complex dependencies, there may be some dependencies that fail to install when installing the dependencies or other reasons that prevent the application from working properly, how do I redeploy it?

It would be nice to have a visual redeployment button like Vercel, so that you can click on redeployment from within the browser, which is very convenient
![image](https://user-images.githubusercontent.com/48512251/192517557-48669b7f-3018-4319-8f70-520f399ded98.png)

**Comment by AyushSehrawat** [comment]
You just edit the things out which gave the error and do `deta deploy` again..

The process is something like you create the micro with `deta new <some options here` , and then a `.deta` folders get created. So to deploy again, you just do `deta deploy` or there is another command `deta watch` to automatically deploy with changes to the files.
For more look https://docs.deta.sh/docs/cli/commands#deta-watch

For CLI commands - https://docs.deta.sh/docs/cli/commands/

**Reply by lete114** [reply]
It does not seem to be in effect
![image](https://user-images.githubusercontent.com/48512251/192523439-716ded96-abc3-477f-8928-a09a72825948.png)

**Reply by lete114** [reply]
use `deta deploy`

```sh
$ deta deploy
error in get latest version tag:  Get "https://api.github.com/repos/deta/deta-cli/releases/latest": read tcp 192.168.8.118:63616->20.205.243.168:443: wsarecv: An existing connection was forcibly closed by the remote host.
Everything up to date
```

**Reply by AyushSehrawat** [reply]
It won't deploy if there are no changes in content of files... Just like git commit...Try adding a space or something somewhere and then deploy

**Reply by AyushSehrawat** [reply]
> It does not seem to be in effect ![image](https://user-images.githubusercontent.com/48512251/192523439-716ded96-abc3-477f-8928-a09a72825948.png)

It should be working fine :thonk: Are you sure it's not something like cache?

**Reply by lete114** [reply]
> It should be working fine :thonk: Are you sure it's not something like cache?

I'm sure it's not a cache.

**Comment by lete114** [comment, marked as answer]
I found the problem, deta doesn't reinstall the dependencies, it reuses the previous ones, it only replaces your code unless you update the dependency version

I have a suggestion, can either make the deployed project completely re-update the dependencies, or add a new parameter after the `deta deploy` command, similar to `deta deploy --no-cache`, while if you don't write it, the caching policy is used by default
